src/main/java/org/apache/commons/net/ftp/FTPFile.java

```diff
@@ -330,13 +330,14 @@ public void setName(final String name) {
     }
 
     /**
-     * Sets if the given access group (one of the <code> _ACCESS </code> constants) has the given access permission (one of the <code> _PERMISSION </code>
+     * Sets if the given access group (one of the {@code _ACCESS} constants) has the given access permission (one of the {@code _PERMISSION}
      * constants) to the file.
      *
-     * @param access     The access group (one of the <code> _ACCESS </code> constants)
-     * @param permission The access permission (one of the <code> _PERMISSION </code> constants)
-     * @param value      True if permission is allowed, false if not.
+     * @param access     The access group (one of the {@code _ACCESS} constants)
+     * @param permission The access permission (one of the {@code _PERMISSION} constants)
+     * @param value      {@code true} if permission is allowed, {@code false} if not.
      * @throws ArrayIndexOutOfBoundsException if either of the parameters is out of range
+     * @throws NullPointerException if you're trying to set permission of an invalid file
```

Comment:
How can an NPE happen here?

Reply:
[FTPListParseEngine](https://github.com/apache/commons-net/blob/master/src/main/java/org/apache/commons/net/ftp/FTPListParseEngine.java) will use `FTPFile(String)` constructor that sets `permissions` to `null`;